pennylane/io.py

```diff
@@ -410,113 +407,140 @@ def from_qiskit_op(qiskit_op, params=None, wires=None):
 
-        The current default behaviour of removing measurements in the QASM code is deprecated
-        and will be changed in a future release. Starting in version ``0.38``, ``from_qasm``
-        will keep the measurements from the QASM code by default. To remove all measurements,
-        set ``measurements=[]`` which overrides the existing measurements with an empty list.
+        qasm_code = 'OPENQASM 2.0;'\\
+                    'include "qelib1.inc";'\\
+                    'qreg q[2];'\\
+                    'creg c[2];'\\
+                    'h q[0];'\\
+                    'measure q[0] -> c[0];'\\
+                    'rz(0.24) q[0];'\\
+                    'cx q[0], q[1];'\\
+                    'measure q -> c;'
 
-    **Example:**
+        loaded_circuit = qml.from_qasm(qasm_code)
 
-    .. code-block:: python
+    >>> print(qml.draw(loaded_circuit)())
+    0: ──H──┤↗├──RZ(0.24)─╭●──┤↗├─┤
+    1: ───────────────────╰X──┤↗├─┤
+
+    Calling the quantum function returns a tuple containing the mid-circuit measurements and the terminal measurements.
+
+    >>> loaded_circuit()
+    (MeasurementValue(wires=[0]),
+    MeasurementValue(wires=[0]),
+    MeasurementValue(wires=[1]))
+
+    A list of measurements can also be passed directly to ``from_qasm`` using the ``measurements`` argument, making it possible to create a PennyLane circuit with :class:`qml.QNode <pennylane.QNode>`.
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit")
+            measurements = [qml.var(qml.Y(0))]
+            circuit = qml.QNode(qml.from_qasm(qasm_code, measurements = measurements), dev)
 
-        >>> hadamard_qasm = 'OPENQASM 2.0;' \\
-        ...                 'include "qelib1.inc";' \\
-        ...                 'qreg q[1];' \\
-        ...                 'creg c[1];' \\
-        ...                 'h q[0];'
-        >>> my_circuit = qml.from_qasm(hadamard_qasm)
-        >>> my_circuit()
+        >>> print(qml.draw(circuit)())
+        0: ──H──┤↗├──RZ(0.24)─╭●─┤  Var[Y]
+        1: ───────────────────╰X─┤
+
+    .. details::
+        :title: Removing terminal measurements
+
+        To remove all terminal measurements, set ``measurements=[]``. This removes the existing terminal measurements and keeps the mid-circuit measurements.
+
+        .. code-block:: python
+
+            loaded_circuit = qml.from_qasm(qasm_code, measurements=[])
+
+        >>> print(qml.draw(loaded_circuit)())
+        0: ──H──┤↗├──RZ(0.24)─╭●─┤
+        1: ───────────────────╰X─┤
+
+        Calling the quantum function returns the same empty list that we originally passed in.
+
+        >>> loaded_circuit()
         []
 
-    The measurements can also be passed directly to the function when creating the
-    quantum function, making it possible to create a PennyLane circuit with
-    :class:`qml.QNode <pennylane.QNode>`:
+        Note that mid-circuit measurements are always applied, but are only returned when ``measurements=None``.
```

Comment:
It might be worth adding an example of how to use this a QNode as well - i.e. something like:

```
dev = qml.device("default.qubit")

@qml.qnode(dev)
def circuit():
    loaded_circuit()
    return qml.expval(qml.Z(1))

circuit()
```

Reply:
I have added one, lmk what you think @lillian542